common-rest/src/test/java/org/uniprot/api/rest/download/heartbeat/HeartBeatProducerTest.java

```diff
@@ -0,0 +1,105 @@
+package org.uniprot.api.rest.download.heartbeat;
+
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.ArgumentCaptor;
+import org.mockito.Captor;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.junit.jupiter.MockitoExtension;
+import org.uniprot.api.rest.download.configuration.AsyncDownloadHeartBeatConfiguration;
+import org.uniprot.api.rest.download.model.DownloadJob;
+import org.uniprot.api.rest.download.repository.DownloadJobRepository;
+
+import java.time.LocalDateTime;
+import java.util.List;
+
+import static org.junit.jupiter.api.Assertions.*;
+import static org.mockito.Mockito.*;
+
+@ExtendWith(MockitoExtension.class)
+class HeartBeatProducerTest {
+    public static final String JOB_ID = "downloadJobId";
+    private final DownloadJob downloadJob = DownloadJob.builder().id(JOB_ID).build();
+    @Mock
+    private AsyncDownloadHeartBeatConfiguration asyncDownloadHeartBeatConfiguration;
+    @Mock
+    private DownloadJobRepository jobRepository;
+    @Captor
+    private ArgumentCaptor<DownloadJob> downloadJobArgumentCaptor;
+    @InjectMocks
+    private HeartBeatProducer heartBeatProducer;
+
+    @Test
+    void create_whenHeartBeatEnabled() {
```

Comment:
Can you please test other use cases as well? with Interval smaller, equals and bigger that batch size